### PR TITLE
SAK-43759 when in admin site, retrieve all roles instead of just the "admin" role

### PR DESCRIPTION
**Pull request by ottenhoff**

### CODE REVIEW

File: sitestats/sitestats-tool/src/java/org/sakaiproject/sitestats/tool/wicket/pages/ReportsEditPage.java [hunk 1]
```diff
@@ -1194,7 +1202,7 @@ private boolean validReportParameters() {
 
 		// check WHO
 		if(getReportParams().getWho().equals(ReportManager.WHO_ROLE)){
-			if(site.getUsersHasRole(getReportParams().getWhoRoleId()).isEmpty())
+			if(!siteId.equals("!admin") && site.getUsersHasRole(getReportParams().getWhoRoleId()).isEmpty())
```
Comment (ern) [inline sitestats/sitestats-tool/src/java/org/sakaiproject/sitestats/tool/wicket/pages/ReportsEditPage.java:1205]: exclude admin workspace as well "~admin"?

File: sitestats/sitestats-tool/src/java/org/sakaiproject/sitestats/tool/wicket/pages/ReportsEditPage.java [hunk 2]
```diff
@@ -1119,16 +1119,24 @@ public IModel getModel(Object value) {
 	}
 	
 	private List<String> getRoles() {
+		Set<String> siteIdWithRoles = new HashSet<>(Arrays.asList(siteId));
+
+		if ("!admin".equals(siteId)) {
```
Comment (ern) [inline sitestats/sitestats-tool/src/java/org/sakaiproject/sitestats/tool/wicket/pages/ReportsEditPage.java:1124]: Also add "~admin" my workspace?

Comment (ottenhoff) [reply sitestats/sitestats-tool/src/java/org/sakaiproject/sitestats/tool/wicket/pages/ReportsEditPage.java]: agree

File: sitestats/sitestats-tool/src/java/org/sakaiproject/sitestats/tool/wicket/pages/ReportsEditPage.java (hunk 1 above)
Comment (ottenhoff) [reply sitestats/sitestats-tool/src/java/org/sakaiproject/sitestats/tool/wicket/pages/ReportsEditPage.java]: agreed

**Pull request merged**